SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -2151,212 +2151,133 @@ void CNEMOEulerSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container
+  so all flow variables can be imposed at the inlet.
+  First, retrieve the specified values for the primitive variables. ---*/
 
+ const string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+ const su2double Temperature = config->GetInlet_Temperature(Marker_Tag);
+ const su2double Pressure = config->GetInlet_Pressure(Marker_Tag);
+ const su2double* Velocity = config->GetInlet_Velocity(Marker_Tag);
+ const su2double* Mass_Frac = config->GetInlet_MassFrac(); 
+ su2double Temperature_ve   = config->GetInlet_Temperature_ve();
+
+ /*--- If no Tve value specified (left as 0 K default), set to Ttr value ---*/
+ if (Temperature_ve == 0.0) {Temperature_ve = Temperature;}
+
+ /*--- Set mixture state ---*/
+ FluidModel->SetTDStatePTTv(Pressure, Mass_Frac, Temperature, Temperature_ve);
+
+ /*--- Compute Ma vector for flow direction ---*/
+ const su2double soundspeed = FluidModel->ComputeSoundSpeed();
+
+ su2double* Mvec = new su2double[nDim];
+
+ unsigned short iDim;
+ for (iDim = 0; iDim < nDim; iDim++)
+   Mvec[iDim] = Velocity[iDim] / soundspeed;
+
+ /*--- Allocate inlet node to compute gradients for numerics ---*/
+ CNEMOEulerVariable node_inlet(Pressure, Mass_Frac, Mvec,
+                                        Temperature, Temperature_ve,
+                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);
+
+ node_inlet.SetPrimVar(0, FluidModel);
+ 
+ su2double Normal[MAXNDIM] = {0.0};
+ 
+ /*--- Loop over all the vertices on this boundary marker ---*/
+ for(unsigned long iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+   unsigned long iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+   /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
+   if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+   /*--- Index of the closest interior node ---*/
+   unsigned long Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+   /*--- Normal vector for this vertex (negate for outward convention) ---*/
+   geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
+   for (iDim = 0; iDim < nDim; iDim++) Normal[iDim] = -Normal[iDim];
+
+   /*--- Set various quantities in the solver class ---*/
+   conv_numerics->SetNormal(Normal);
+   conv_numerics->SetConservative(nodes->GetSolution(iPoint), node_inlet.GetSolution(0));
+   conv_numerics->SetPrimitive(nodes->GetPrimitive(iPoint), node_inlet.GetPrimitive(0));
+
+   /*--- Pass supplementary info to CNumerics ---*/
+   conv_numerics->SetdPdU(nodes->GetdPdU(iPoint), node_inlet.GetdPdU(0));
+   conv_numerics->SetdTdU(nodes->GetdTdU(iPoint), node_inlet.GetdTdU(0));
+   conv_numerics->SetdTvedU(nodes->GetdTvedU(iPoint), node_inlet.GetdTvedU(0));
+   conv_numerics->SetEve(nodes->GetEve(iPoint), node_inlet.GetEve(0));
+   conv_numerics->SetCvve(nodes->GetCvve(iPoint), node_inlet.GetCvve(0));
+
+   bool dynamic_grid  = config->GetGrid_Movement();
```

Comment:
everything that can be const should be const
```suggestion
   const bool dynamic_grid  = config->GetGrid_Movement();
```